app/models/host/managed.rb

```diff
@@ -29,7 +29,11 @@ class Host::Managed < Host::Base
   has_many :all_reports, :foreign_key => :host_id
 
   belongs_to :image
-  has_many :host_statuses, -> { where.not(type: nil) }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :destroy
+  if defined? ForemanHostReports
+    has_many :host_statuses, -> { where("type != 'HostStatus::ConfigurationStatus' and type is not null") }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :delete_all
+  else
+    has_many :host_statuses, -> { where.not(type: nil) }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :destroy
+  end
```

Comment:
Had to learn the hard way that it is not possible to override Ruby on Rails associations from plugins. At least it does not work for my case, not sure why.

Reply:
@ezr-ondrej can you show me how to do this from plugin?

Reply:
Ondrej says it is very hacky to do from plugin so the agreement is to keep it this way.

Reply:
There might be way actually to use default_scope in HostStatus. Let me test this.

Reply:
I tried to implement this via default scope on HostReportStatus but unfortunately this does not work. That model does use STI so I cannot limit what other associations see. I think if statement is the only option if we don't want to do an ugly hack like in Puppet.